iib/workers/tasks/opm_operations.py

```diff
@@ -1104,3 +1116,72 @@ def deprecate_bundles(
         cmd.append(container_tool)
     with set_registry_token(overwrite_target_index_token, from_index):
         run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles')
+
+
+def _find_index_version(index_image: str) -> str:
+    """
+    Find the OCP version of the given index image.
+
+    :param str index_image: index_image pull spec
+    :return: version of index image
+    :rtype: str
+
+    """
+    from iib.workers.tasks.utils import get_image_label
+
+    index_image_version = index_image.split(":")[1]
```

Comment:
I don't think we should rely on the `tag` to get the OCP version of the index image. Only the label should be considered unless there's a use case for using the tag. Even if there is a use case, the first choice should be the label, no?

Reply:
The reason I did not take label as first choice was that from the tag we just need to fetch the version, whereas from the label, we inspect the image and then get the label so just pure operational steps wise, tag was first preference

Reply:
right, but using the tag for getting the index version is unreliable and error prone. Nothing stops a user from specifying the tag as `v4.<random-string>` for a v4.17 index and then IIB would use the default OPM (which might not be the correct one)

Reply:
I am here with Yash. We should use label as the reference for index image version. Do not worry about performance. It will be fast since there is memcache in place and we get this info on multiple places so we are fetching the data only once.

Reply:
Yup, already removed the function